docs/versions/latest/Components/Archetypes/typescript/General/Known Issues.md

```diff
@@ -0,0 +1,112 @@
+# Known Issues
+A list of known typescript archetype issues
+
+## Table Of Contents
+1. [Array functions are not transpiled to vRO code](#array-functions-are-not-transpiled-to-vro-code)
+
+### Array functions are not transpiled to vRO code
+
+If an `Array` variable is not explicitly defined or recognised as such, the transpiler does not convert TS specific `Array` prototype functions (`find()`, `some()`, etc.) into vRO compatible code resulting in a runtime error.
+
+Consider the following example:
+```javascript
+const testArray = [1, 2, 3, 4, 5];
+
+let objectsForIteration = null;
+if (true) {
+    objectsForIteration = testArray;
+}
+
+// Fails to transpile correctly because type is resolved to "any"
+const res = objectsForIteration.find(o => o === 2)
+System.log(res + "")
+```
+
+The code above is be converted to the following vRO code, which during execution throws: `TypeError: Cannot find function find in object 1,2,3,4,5.`
+
+```javascript
+var testArray = [1, 2, 3, 4, 5];
+
+var objectsForIteration = null;
+if (true) {
+    objectsForIteration = testArray;
+}
+
+// Fails to transpile correctly because type is resolved to "any"
+var res = objectsForIteration.find(function (o) { return o === 2; });
+System.log(res + "");
+```
+
+Proper variable typisation solves this problem. Let's revisit the example but this time we explicitly define what type of values we expect the `objectsForIteration` variable to receive.
+
+```javascript
+const testArray = [1, 2, 3, 4, 5];
+
+let objectsForIteration: Array<number> = null;
+if (true) {
+    objectsForIteration = testArray;
+}
+
+// Transpiles correctly because of explicit typisation
+const res = objectsForIteration.find(o => o === 2)
+System.log(res + "")
+```
+
+The code is transpiled correctly to vRO code and executes successfully.
+
+```javascript
+var __global = System.getContext() || (function () {
+    return this;
+}).call(null);
+var VROES = __global.__VROES || (__global.__VROES = System.getModule("com.vmware.pscoe.library.ecmascript").VROES());
+var testArray = [1, 2, 3, 4, 5];
+
+var objectsForIteration = null;
+if (true) {
+    objectsForIteration = testArray;
+}
+// Transpiles correctly because of explicit typisation
+var res = VROES.Shims.arrayFind(objectsForIteration, function (o) { return o === 2; });
+System.log(res + "");
+```
+Recommended configuration to prevent such issues is setting `strictNullChecks` to `true` in your project's local `tsconfig.json`. This allows for type hint warning to be displayed in such case.
```

Comment:
Also, it might be a good idea to separate the rest of the changes in this file in a sub-section, something like:
#### How to prevent this error

The reason for this being that the information suggests a way to avoid hitting this problem and not so much a description of what the problem is.